src/fdcache_entity.h

```diff
@@ -135,9 +137,12 @@ class FdEntity
         }
         int GetOpenCountHasLock() const REQUIRES(FdEntity::fdent_lock);
         // TODO: should this require a lock?
-        const std::string& GetPath() const { return path; }
+        std::string GetPath() const {
+            const std::lock_guard<std::mutex> lock(fdent_lock);
```

Comment:
This fix is ​​fine, but it causes a deadlock in rename_object() by locking fdent_lock here. (A fix to s3fs.cpp is also required.)